perception/autoware_tensorrt_bevdet/src/gatherbev_plugin.cu

```diff
@@ -0,0 +1,308 @@
+  const void * const * inputs, void * const * outputs, void * workspace,
+  cudaStream_t stream) noexcept
+{
+  // input[0] == adj_feat   b x 8 x 80 x 128 x 128
+  // input[1] == curr_feat  b x 80 x 128 x 128
+  // input[2] == flag       b x 1
+  // out[0]                 b x (8+1)*80 x 128 x 128
+
+  // int nthreads, // b * (adj_num + 1) * map_size
+  // int adj_num,
+  // int channel,
+  // int map_size,
+
+  // int flag = 0;
+  // CHECK_CUDA(cudaMemcpy(&flag, inputs[2], sizeof(int), cudaMemcpyDeviceToHost));
+
+  int b = inputDesc[0].dims.d[0];
+  int adj_num = inputDesc[0].dims.d[1];
+  int map_size = inputDesc[0].dims.d[3] * inputDesc[0].dims.d[4];
+  int channel = inputDesc[0].dims.d[2];
+
+  int feat_step = inputDesc[1].dims.d[1] * inputDesc[1].dims.d[2] * inputDesc[1].dims.d[3];
+
+  int nthreads = b * (adj_num + 1) * map_size;
+
+  dim3 grid(GET_BLOCKS(nthreads));
+  dim3 block(NUM_THREADS);
+
+  switch (int(outputDesc[0].type)) {
+    case int(DataType::kFLOAT):
+      // fp32
+      copy_feat_kernel<<<grid, block, 0, stream>>>(
+        nthreads, adj_num, channel, map_size, reinterpret_cast<const float *>(inputs[0]),
+        reinterpret_cast<const float *>(inputs[1]), reinterpret_cast<const int *>(inputs[2]),
+        reinterpret_cast<float *>(outputs[0]));
+
+      break;
+    case int(DataType::kHALF):
+      // fp16
+      copy_feat_kernel<<<grid, block, 0, stream>>>(
+        nthreads, adj_num, channel, map_size, reinterpret_cast<const __half *>(inputs[0]),
+        reinterpret_cast<const __half *>(inputs[1]), reinterpret_cast<const int *>(inputs[2]),
+        reinterpret_cast<__half *>(outputs[0]));
+      break;
+    default:  // should NOT be here
+      RCLCPP_ERROR(rclcpp::get_logger("GatherBEVPlugin"), "\tUnsupport datatype!");
+  }
+  return 0;
+}
+
+void GatherBEVPlugin::destroy() noexcept
+{
+  delete this;
+  return;
+}
+
+int32_t GatherBEVPlugin::initialize() noexcept
+{
+  return 0;
+}
+
+void GatherBEVPlugin::terminate() noexcept
+{
+  return;
+}
+
+size_t GatherBEVPlugin::getSerializationSize() const noexcept
+{
+  return sizeof(m_);
+}
+
+void GatherBEVPlugin::serialize(void * buffer) const noexcept
+{
+  memcpy(buffer, &m_, sizeof(m_));
+  return;
+}
+
+void GatherBEVPlugin::setPluginNamespace(const char * pluginNamespace) noexcept
+{
+  namespace_ = pluginNamespace;
+  return;
+}
+
+const char * GatherBEVPlugin::getPluginNamespace() const noexcept
+{
+  return namespace_.c_str();
+}
+
+const char * GatherBEVPlugin::getPluginType() const noexcept
+{
+  return GATHERBEV_PLUGIN_NAME;
+}
+
+const char * GatherBEVPlugin::getPluginVersion() const noexcept
+{
+  return GATHERBEV_PLUGIN_VERSION;
+}
+
+void GatherBEVPlugin::attachToContext(
+  cudnnContext * contextCudnn, cublasContext * contextCublas, IGpuAllocator * gpuAllocator) noexcept
+{
+  return;
+}
+
+void GatherBEVPlugin::detachFromContext() noexcept
+{
+  return;
+}
+
+// class GatherBEVPluginCreator
```

Comment:
Similar to the `GatherBEVPlugin` class, the `GatherBEVPluginCreator` class should also be enclosed within the `autoware_tensorrt_bevdet` namespace for better code organization and to avoid potential namespace collisions.